modules/data_release/ajax/AddPermission.php

```diff
@@ -12,25 +12,84 @@
  *  @link     https://github.com/aces/Loris
  */
 
-$DB =& Database::singleton();
+$user     = User::singleton();
+$factory  = NDB_Factory::singleton();
+$settings = $factory->settings();
+$baseURL  = $settings->getBaseURL();
+$DB       = $factory->database();
 
-if ($_POST['action'] == 'addpermission') {
-    $userid          = $_POST['userid'];
-    $data_release_id = $_POST['data_release_id'];
-    $success         = $DB->insert(
-        'data_release_permissions',
-        array(
-         'userid'          => $userid,
-         'data_release_id' => $data_release_id,
-        )
-    );
+if ($_POST['action'] == 'addpermission' && $user->hasPermission('superuser')) {
+    if (!empty($_POST['data_release_id']) && empty($_POST['data_release_version'])) {
+        $userid          = $_POST['userid'];
+        $data_release_id = $_POST['data_release_id'];
+        $success         = $DB->insert(
+            'data_release_permissions',
+            array(
+             'userid'          => $userid,
+             'data_release_id' => $data_release_id,
+            )
+        );
+    } elseif (empty($_POST['data_release_id'])
+        && !empty($_POST['data_release_version'])
+    ) {
+        $userid = $_POST['userid'];
+        $data_release_version = $_POST['data_release_version'];
 
-    $factory  = NDB_Factory::singleton();
-    $settings = $factory->settings();
+        $IDs = $DB->pselect(
+            "SELECT id FROM data_release WHERE "
+            . "version=:data_release_version",
+            array('data_release_version' => $data_release_version)
+        );
 
-    $baseURL = $settings->getBaseURL();
+        foreach ($IDs as $ID) {
```

Comment:
i don't know much about the business logic behind data release, but is it not problematic to give the user permission for all data releases with matching versions?

Reply:
I've intentionally kept everything as granular as possible just because I can't anticipate what kind of permission structure a study would want on their files...and I'm considering the scenario let's say a study tags the Data Use Agreement file as Version 2 and the .zip Data Release file as Version 2 as well(since they're both Version 2...) but first assigns everyone the Data Use Agreement file to download and sign the form before granting them the .zip Data Release 🤷‍♂️